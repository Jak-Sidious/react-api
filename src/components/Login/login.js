import React from "react";
import {Button, FormGroup, FormControl} from "react-bootstrap";
import axios from 'axios';

const ROOT_URL = 'http://127.0.0.1:5000/apiv1/';
const LOGIN_URL = 'users/login';
class Login extends React.Component{
  constructor(props){
    super(props);

    this.state = {
      signIn : {
        username: '',
        password: ''
      }

    };
  }
  handleChange = (event) =>{
    const {name, value} = event.target;
    this.setState({[name]: value});
  }


  handleLogin = (event) =>{
    event.preventDefault();
    const user = {
      username: this.state.signIn.username,
      password: this.state.signIn.password
    }
    console.log(this.state);
    axios.post(`${ROOT_URL}${LOGIN_URL}`, user)
    .then((response) => {
      this.props.history.push('/landing');
      console.log(response);
    });
  }

  render() {
    return(
      <div className="logBackground">
        <div className="wrapper">
          <form
            className="form-signin"
            onSubmit={this.handleLogin}>
            <h2 className="form-signin-heading">Please login</h2>
            <FormGroup
              controlid="formBasicText"
            >
              <FormControl className="nameField"
                autoFocus
                name="username"
                type="username"
                onChange={this.handleChange}
                placeholder="Enter Username"
              />
              <FormControl className="passwordField"
                autoFocus
                name="password"
                type="password"
                onChange={this.handleChange}
                placeholder="Enter Username"
              />
            <br/>
              <Button
                type="submit"
                bsStyle="success"
                bsSize="large"
                block>
                Login
              </Button>

              <FormControl.Feedback />
            </FormGroup>
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
