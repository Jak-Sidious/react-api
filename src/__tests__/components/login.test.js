import React from 'react';
import { shallow } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import Login from '../../components/Login/login';

describe('Login component', () => {
  const login = jest.fn();
  const wrapper = shallow(<Login login={login} location={{}} />);
  const preventDefault = jest.fn();

  it('renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('it renders state initially', () => {
    expect(wrapper.state().username).toEqual('');
    expect(wrapper.state().password).toEqual('');
  });

  it('should login a user', () => {
    wrapper.setState({
      username: 'user',
      password: 'user1234'
    });
    wrapper.find('#btn1').simulate('submit', { preventDefault });
    expect(wrapper.instance().handleLogin({ preventDefault }));
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has clickable button which logs in a user', () => {
    expect(wrapper.find('#btn1').length).toBe(1);
    expect(wrapper.find('#btn1').simulate('click'));
  });
});
