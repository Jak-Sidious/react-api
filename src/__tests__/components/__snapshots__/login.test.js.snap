// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly without crashing 1`] = `
<div
  className="logBackground"
>
  <_class
    options={Object {}}
  />
  <div
    className="wrapper"
  >
    <Form
      as="form"
      className="form-signin"
      onSubmit={[Function]}
    >
      <h2
        className="form-signin-heading"
      >
        Login Form
      </h2>
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="nameField"
        control={[Function]}
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        type="username"
      />
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="passwordField"
        control={[Function]}
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
      />
      <Button
        as="button"
        className="left floated"
        color="blue"
        id="btn1"
        size="large"
        type="submit"
        value="submit"
      >
        Login
      </Button>
      <Button
        action="/login"
        as={[Function]}
        className="right floated"
        color="green"
        id="btn"
        size="large"
        to="/"
      >
        Register
      </Button>
      <br />
    </Form>
  </div>
</div>
`;

exports[`Login component should login a user 1`] = `
<div
  className="logBackground"
>
  <_class
    options={Object {}}
  />
  <div
    className="wrapper"
  >
    <Form
      as="form"
      className="form-signin"
      onSubmit={[Function]}
    >
      <h2
        className="form-signin-heading"
      >
        Login Form
      </h2>
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="nameField"
        control={[Function]}
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        type="username"
      />
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="passwordField"
        control={[Function]}
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
      />
      <Button
        as="button"
        className="left floated"
        color="blue"
        id="btn1"
        size="large"
        type="submit"
        value="submit"
      >
        Login
      </Button>
      <Button
        action="/login"
        as={[Function]}
        className="right floated"
        color="green"
        id="btn"
        size="large"
        to="/"
      >
        Register
      </Button>
      <br />
    </Form>
  </div>
</div>
`;
